WordPress/src/main/java/org/wordpress/android/ui/posts/PostListMainViewModel.kt

```diff
@@ -90,4 +124,35 @@ class PostListMainViewModel @Inject constructor(
             }
         }
     }
+
+    sealed class AuthorFilterListItemUIState(
+        internal val text: UiString, @DrawableRes val iconRes: Int, val isSelected: Boolean
+    ) {
+        class Everyone(isSelected: Boolean) : AuthorFilterListItemUIState(
+                text = UiStringRes(R.string.post_list_author_everyone),
+                iconRes = R.drawable.ic_multiple_users_white_24dp,
+                isSelected = isSelected
+        )
+
+        class Me(val avatarUrl: String?, isSelected: Boolean) : AuthorFilterListItemUIState(
+                text = UiStringRes(R.string.post_list_author_me),
+                iconRes = R.drawable.ic_user_circle_grey_24dp,
+                isSelected = isSelected
+        )
+    }
+
+    enum class AuthorFilterSelection {
+        ME, EVERYONE;
+
+        companion object {
+            @JvmStatic
+            fun valueOf(value: String, default: AuthorFilterSelection): AuthorFilterSelection {
+                return try {
+                    valueOf(value)
+                } catch (ignored: IllegalArgumentException) {
+                    default
```

Comment:
My thought process here is that this exception would only every occur if we change the options in the Enum, in which case the user would just have their author filter returned to the default.

Happy to make those changes, although I think crashing in debug might be annoying for developers if we make any enum changes in the future.

Reply:
I see, I think we'd still want to log the exception (I agree we wouldn't want to crash in debug). However, I like your solution way more. Btw does it still make sense to use `.ordinal` when each of the enum values have an `id`?

Reply:
The only places we are using `.ordinal` at the moment are when we set the selection index, and when we update our selection based off the position from the adapter.

We could do away with that if we passed the `id` via the view model and then grab that back out from the adapter to pass back to the view model. Happy to make that change if you think it will make the code simpler. Should I make that change?

Reply:
Thanks for the changes!

>We could do away with that if we passed the id via the view model and then grab that back out from the adapter to pass back to the view model. Happy to make that change if you think it will make the code simpler. Should I make that change?

I'm on the edge. If we stored the ordinal into a persistance storage, I'd definitely vote for removing it and store the id instead. Since we use this just in runtime, I can't think of a scenario in which it could turn against us. Wdyt @oguzkocer ?

Reply:
> Happy to make those changes, although I think crashing in debug might be annoying for developers if we make any enum changes in the future.

I think that's exactly the point. Ideally it'd crash for whoever changes the enum and will come up with the correct solution for it or make it explicit that we are ignoring this problem. It's fine if something crashes for our developers if there is value in it.

> Btw does it still make sense to use .ordinal when each of the enum values has an id?

I think we should use the `id` and not the `ordinal`.

> We could do away with that if we passed the id via the view model and then grab that back out from the adapter to pass back to the view model. Happy to make that change if you think it will make the code simpler. Should I make that change?

Hopefully we don't have to do that and even if we do, we can do it just by passing the `enum` itself and not the `id` of it. I'll check this out in my review.